elements/rh-switch/rh-switch.css

```diff
@@ -0,0 +1,147 @@
+  --_margin-inline: 5px; /* non standard spacer */
+}
+
+#container {
+  display: inline-flex;
+  align-items: center;
+  gap: var(--rh-space-lg, 16px);
+}
+
+:host(:disabled) {
+  pointer-events: none;
+  cursor: not-allowed;
+
+  --_switch-track-background-color: var(--rh-color-gray-30, #c7c7c7);
+}
+
+[hidden] {
+  display: none !important;
+}
+
+#switch {
+  align-items: center;
+  display: inline-flex;
+  flex-shrink: 0;
+  position: relative;
+  width: var(--_switch-width);
+  height: var(--_switch-height);
+  border-radius: var(--rh-border-radius-pill, 64px);
+}
+
+:host(:is(:focus, :focus-within)) #switch {
+  outline:
+    var(--rh-border-width-md, 2px)
+    solid
+    var(--_switch-focus-outline-color, var(--rh-color-border-interactive-on-light, #0066cc));
+  outline-offset: var(--rh-space-sm, 6px);
+}
+
+/* track */
+#switch:before {
+  content: "";
+  display: flex;
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  border-radius: inherit;
+  justify-content: center;
+  align-items: center;
+  background-color: var(--_switch-track-background-color);
+}
+
+/* handle */
+#switch:after {
+  content: "";
+  height: var(--_switch-handle-size);
+  width: var(--_switch-handle-size);
+  border-radius: var(--rh-border-radius-pill, 64px);
+  transform-origin: center;
+  z-index: 0;
+  background-color: var(--_switch-handle-color, var(--rh-color-surface-lightest, #ffffff));
+  margin-inline: var(--_margin-inline);
+  translate: 0%;
+  transition: translate 0.25s ease 0s;
+}
+
+:host([checked]:not(:disabled)) {
+  --_switch-track-background-color:
+    var(
+        --_switch-checked,
+        var(--rh-color-accent-base-on-light, #0066cc)
+      );
+}
+
+:host([checked]) #switch:after {
+  translate:
+    calc(
+        var(--_switch-width) - (var(--_switch-handle-size) +
+        (var(--_margin-inline) * 2))
+      );
+}
+
+:host([checked]) .rtl #switch:after {
+  translate:
+    calc(
+        -1 * (
+          var(--_switch-width) - (var(--_switch-handle-size) +
+          (var(--_margin-inline) * 2))
+        )
+      );
+}
+
+svg {
+  fill: currentcolor;
+  margin-inline: var(--_margin-inline);
+  font-size: var(--rh-font-size-body-text-xs, 0.75rem);
+  color: var(--_switch-handle-color, var(--rh-color-icon-secondary-on-dark, #ffffff));
+  position: absolute;
+  z-index: 1;
+}
+
+:host(:not([checked])) ::slotted(*) {
+  color: var(--rh-color-text-secondary-on-light, #4d4d4d);
+}
+
+:host(:disabled) ::slotted(*),
+:host(:disabled) slot[name^="message"] span[aria-hidden] {
+  color: var(--rh-color-gray-50, #707070);
+}
+
+.dark {
+  --_switch-track-background-color: var(--rh-color-gray-40, #a3a3a3); /* no disabled token yet */
+  --_switch-handle-color: var(--rh-color-surface-dark-alt, #292929);
+}
+
+.dark:after {
+  box-shadow: var(--rh-box-shadow-sm, 0 2px 4px 0 rgba(21, 21, 21, 0.2));
+}
+
+:host(:disabled) .dark {
+  --_switch-track-background-color: var(--rh-color-gray-60, #4d4d4d);
```

Comment:
The text color for dark disabled switches with labels should be `--rh-color-gray-40`. Everything else looks great! 

@coreyvickery, I created [an issue](https://github.com/RedHat-UX/red-hat-design-system/issues/1710) to update switch colors in the Figma library.

Reply:
@marionnegp The value I see in figma for a disabled track on dark is: `#383838`
![Screenshot 2024-07-11 at 5 03 45 PM](https://github.com/RedHat-UX/red-hat-design-system/assets/13666/23d81712-61f9-4bba-99c6-38b3282a5556)

This isn't the color you are wanting to change in the issue you created is it?

Changing the dark disabled track background color to gray-40 would make the track the same color as when its not disabled:
![image](https://github.com/RedHat-UX/red-hat-design-system/assets/13666/c140e13a-94c5-432c-b023-9e6577e29d14)

The property @bennypowers and I were discussing above was the text disabled color which was different based on which variant of dark it was on.  `#A3A3A3` as its in figma does not work on `dark` but does on `darker` `darkest`.  Because in figma we are only representing switch on one dark variant it was not caught that the text value isn't accessible on the `dark` variant background.  

I had added a work around for this to apply a different text color, but @bennypowers suggestion was to use the value from `dark` across all instead which is: `#4d4d4d` and I believe is a better compromise then the work around.  

Let me know if you want to meet about this to clarify, if I am off base here.

Reply:
No, `#4d4d4d` for the track color is correct here. I have an [issue](https://github.com/RedHat-UX/red-hat-design-system/issues/1710) to fix the dark, disabled switch in Figma based on what we have in the repo.

I was thinking about changing the text color for dark, disabled to `var(--rh-color-gray-40, #A3A3A3)`. It has more contrast than the current text color of `#707070` but still looks more subtle than the text color for enabled switches. This should be good for all dark color palettes. What do you think? (Also, sorry I missed the messages in a previous thread about the track color!!)